### PR TITLE
Add china northwest driver about aws

### PR DESCRIPTION
**Pull request by yangkang55**
### Description
in order to use this in chia ningxia province  s3 service ,so 
Add china northwest driver about aws
Add china northwest type about aws
Add china northwest providers about aws
### Status

done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by vdloo** [comment]
You could add a test [here](https://github.com/apache/libcloud/blob/bad71c41ece620a1d7d957eb0fee66f2787e66fa/libcloud/test/storage/test_s3.py#L1011), but that only exists for a couple of the currently available S3StorageDrivers anyway so I'm not sure if that's a big deal.

File: libcloud/storage/providers.py [hunk 1]
```diff
@@ -37,6 +37,8 @@
     ('libcloud.storage.drivers.s3', 'S3USGovWestStorageDriver'),
     Provider.S3_CN_NORTH:
     ('libcloud.storage.drivers.s3', 'S3CNNorthStorageDriver'),
+    Provider.S3_CN_NORTHQWEST:
```
Comment (vdloo) [inline libcloud/storage/providers.py]: Was this Q intended? Or did you perhaps mean `S3_CN_NORTHWEST` instead of `S3_CN_NORTHQWEST` here?

Comment (yangkang55) [reply libcloud/storage/providers.py]: sorry, it‘s spelling mistakes，i will add the test and check my code repeatedly；
And, in my company, the s3 service is in cn-northwest region, the technicist of aws told us  servers  in this region will raise continuously. without that change,  using libcloud to access s3 object in cn-northwest won't work. so this is a important deal for us! Sorry for my mistake again!

**Comment by vdloo** [comment]
@yangkang55 Thanks! Merged in trunk. :sparkles:

**Comment by pquentin** [comment]
Thanks @vdloo!

**Pull request closed without merging**